### PR TITLE
Authentication/Login/Logout and editing of user info

### PR DESCRIPTION
**Pull request by noodaj**
## Description / Changes Made

-  Added Clerk
     - Added login
     - Added logout
     - Added register
     - Added a way to update the user's information if available. (If the user uses an OAuth, we will not be able to update their profile picture, first or last name) 
- Created input component that includes React Hook Forms form validation for better input checking with Zod validation
- Changed tab layout so that the header is now not shown 
- Added blank text to remaining pages like `friends/feed`
- Removed unused components from `components/`
- Added initialization of Firebase

## How to Test
1. None

## Checklist

- [x] I have added/updated relevant documentation, and I have followed the coding style guidelines.
- [x] I have added/updated tests, and I have run the test suite and all tests pass.
- [x] I have checked for any potential conflicts with other branches and fixed any merge conflicts.

### CODE REVIEW

File: frontend/app/(tabs)/_layout.tsx [hunk 1]
```diff
@@ -33,6 +21,7 @@ export default function TabLayout() {
           tabBarIcon: ({ color, size }) => (
             <Entypo name={'home'} color={color} size={size} />
           ),
+          headerShown: false,
```
Comment (mitshelle) [inline frontend/app/(tabs)/_layout.tsx:24]: I like this change! Gives more real estate overall. 

However, other developers on the front end are going to have to adjust their pages they are working on. Just let them know in the meeting and get feedback! :)

File: frontend/components/UserInput.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+interface UserInputProps<T> {
+  labelID: string;
+  placeholder: string;
+  useLabel?: boolean;
+  field: ControllerRenderProps<T, any>;
+  sx?: Record<string, any>;
+}
+
+const UserInput = <T extends any>({
```
Comment (mitshelle) [inline frontend/components/UserInput.tsx:13]: This component will be helpful when we set up queries for the backend!

File: frontend/app/(auth)/register.tsx [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+  });
+  const registerUser: SubmitHandler<RegisterSchemaInputs> = async (data) => {
+    try {
+      const { createdSessionId, status } = await create({
+        username: data.username,
+        password: data.password,
+        firstName: data.firstName ?? '',
+        lastName: data.lastName ?? '',
+      });
+
```
Comment (mitshelle) [inline frontend/app/(auth)/register.tsx:38]: The register user works well. The checks for the password are a good implementation!

File: frontend/app/(tabs)/profile/profileEditModal.tsx [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+  } = useForm<UserState>({
+    defaultValues: {
+      username: user?.username,
+      firstName: user?.firstName,
+      lastName: user?.lastName,
+      password: user?.passwordEnabled ? 'password' : '',
+    },
+  });
+
+  const handleUserProfileChange = async () => {
```
Comment (mitshelle) [inline frontend/app/(tabs)/profile/profileEditModal.tsx:26]: This feature is off to a great start! I like being able to add a profile picture and editing my bio.

File: frontend/types/user.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      'Password must contain at least one uppercase letter, one lowercase letter, one number and one special character.',
+    ),
+  firstName: z.string().optional(),
+  lastName: z.string().optional(),
+});
+
+export type RegisterSchemaInputs = z.infer<typeof RegisterSchema>;
+
+//login
+export const LoginSchema = z.object({
```
Comment (mitshelle) [inline frontend/types/user.ts:32]: These checks are a must need and nice job including them!

**Review by mitshelle — APPROVED** [review]
Amazing job @noodaj! This is off to a really really great start. 

The login and register page works as well as any. It works like a charm. There are a couple bugs I found, but I think those should be addressed in another PR. Look out for the issues I submit! :) 

The profile page is off to a great start. It looks similar to what users would normally see when they navigate to their profile on any other app. I like how I can change my bio and profile picture and it logs it. 
image
image

The stylistic changes are also good, but we just need to let the other frontenders know so they can update their pages. 

Overall, this is a really great base. It will work even better once all the backend is connected. Nice job!

**Comment by rparker2003** [comment]
Jadon, your work on the app is simply outstanding! The login and register pages function seamlessly, and the design is just perfect—it adds a polished touch to the entire app. While a few warnings are coming up with certain components, especially during transitions between dark and light modes on Android, your dedication to detail shines through. The edit modal is a real standout feature, and the client-side validation you've implemented will undoubtedly streamline processes for the backend team, saving them valuable time. Your contributions truly elevate the project, Jadon. Keep up the amazing work!

**Review by rparker2003 — APPROVED** [review]

**Pull request merged**